### PR TITLE
FEAT: Minimal APIs implementation

### LINKED ISSUE #56: FEATURE: Replace Controllers with Minimal APIs implementation
## Detailed Description
Replace the current Controllers implementation with the newer Minimal APIs mechanism.

## Context
The Minimal APIs have much less dependencies and provide improved performance against the Controllers. Also, they are flexible enough to be used in very small API projects, which make it ideal for the default implementation of REST APIs in Monaco.

## Possible Implementation
I would think towards grouping the APIs by their declared resources, similarly to what we do nowadays with routing in Controllers, separating them in different files.

### PR DESCRIPTION
**Pull request by CesarD**
## Description
The classic Controllers implementation is replaced by the new, more modern and performant Minimal APIs implementation.

## Related Issue
Closes #56.

## Motivation and Context
As explained in #56, the Minimal APIs implementation brings improved performance to the request execution because of the reduced amount of dependencies they provide. They are also more flexible to make this a better fit even for smaller projects as they are easy to compose and have a better system to document them in Swagger.

## How Has This Been Tested?
This should be transparent and the endpoints respond exactly as the ones provided by the Controllers.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+namespace Monaco.Template.Backend.Api.Endpoints;
+
+public static class Products
+{
+	public static IEndpointRouteBuilder AddProducts(this IEndpointRouteBuilder builder, ApiVersionSet versionSet)
+	{
+		var products = builder.CreateApiGroupBuilder(versionSet, "Products");
+
+		products.MapGet("",
```
Comment (matthewtoghill) [inline src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs:24]: Perhaps it is worth adding another overload of the `Map{Verb}` extension methods without the `string pattern` parameter that then calls another overload passing `string.Empty` or `""` for it's `pattern` parameter? What do you think?

File: src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+							sender.ExecuteQueryAsync(new GetProductPage.Query(request.Query)),
+						"GetProducts",
+#if (disableAuth)
+						"Gets a page of products");
+#else
+						"Gets a page of products")
+				.AllowAnonymous();
+#endif
+
+		products.MapGet("/{id:guid}",
```
Comment (matthewtoghill) [inline src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs:36]: I don't think it matters much but here we have `/{id:guid}` and elsewhere we have just `{id:guid}`. They both end up the same but may be clearer to only use one format?

File: src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs (hunk 1 above)
Comment (CesarD) [reply src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs:24]: Would you mind to suggest some snippet of code about what you mean?
That would be easier :)

File: src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs (hunk 2 above)
Comment (CesarD) [reply src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs]: Oh, right... I guess it must have been a wild copy/paste 😅 
Fixed.

File: src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs (hunk 1 above)
Comment (CesarD) [reply src/Content/Backend/Solution/Monaco.Template.Backend.Api/Endpoints/Products.cs:24]: Hmmm... I think I would prefer to leave it explicit for the dev to set the empty string there...
When someone takes a quick look at the same position of each endpoint, everyone usually attempts to find the string written there in order to get an idea of what is the actual route that endpoint is pointing too... I think taking it out might make it less obvious and is something where being a little extra verbose could be beneficial for the sake of quick reading the endpoints.

image

There might be more endpoints and having to switch between overloads with and without pattern parameter there, might be like climbing a stair with the steps of different dimensions 😅

**Review by matthewtoghill — APPROVED** [review]

**Review by Gonzo345 — APPROVED** [review]

**Pull request merged**